test/parallel/test-http2-socket-close.js

```diff
@@ -62,6 +62,6 @@ netServer.listen(0, common.mustCall(() => {
       setTimeout(() => {
         serverH2Session.close();
       }, 10);
-    }, 10);
+    }, 1000);
```

Comment:
Another option without increasing the test duration could be adding a noop listener for the `'error'` event on the peer where the error is emitted. What do you think?

Reply:
This is what I mean

```diff
diff --git a/test/parallel/test-http2-socket-close.js b/test/parallel/test-http2-socket-close.js
index 02db77bcf8..ec6c8fe410 100644
--- a/test/parallel/test-http2-socket-close.js
+++ b/test/parallel/test-http2-socket-close.js
@@ -25,6 +25,7 @@ let serverH2Session;
 
 const netServer = net.createServer((socket) => {
   serverRawSocket = socket;
+  socket.on('error', () => {});
   h2Server.emit('connection', socket);
 });
 

```

FWIW I can't reproduce the issue even if `serverRawSocket.destroy()` is called immediately (outside the `setTimeout()` callback).

Reply:
Thank you for the review. 
I can easily reproduce this by using the following command:
`python tools\test.py --repeat=2000 parallel/test-http2-socket-close`

I've applied the above patch locally, but the test is still flaky.

Reply:
does still flaky means that `ECONNRESET` is still unhanded? If so I wonder where that error is emitted.

Reply:
Yes, the error is the same.
From what I've seen from the logs, the following throws the exception:
https://github.com/nodejs/node/blob/6cc4d5fc87503a6bc033b70f125d319d46f1c01e/lib/internal/stream_base_commons.js#L216

Reply:
Yes, but it should not crash the process if there is a listener for the `'error'` event.

Reply:
This should work

```
diff --git a/test/parallel/test-http2-socket-close.js b/test/parallel/test-http2-socket-close.js
index 02db77bcf8..a5adfc8c66 100644
--- a/test/parallel/test-http2-socket-close.js
+++ b/test/parallel/test-http2-socket-close.js
@@ -42,6 +42,8 @@ netServer.listen(0, common.mustCall(() => {
     rejectUnauthorized: false
   });
 
+
+  proxyClient.on('error', () => {});
   proxyClient.on('close', common.mustCall(() => {
     netServer.close();
   }));
@@ -51,6 +53,7 @@ netServer.listen(0, common.mustCall(() => {
     ':path': '/'
   });
 
+  req.on('error', () => {});
   req.on('response', common.mustCall((response) => {
     assert.strictEqual(response[':status'], 200);
 

```

Reply:
OK, this seems to have fixed the issue (it didn't fail in 10k runs). I'll update the PR accordingly. Thanks.

Reply:
@huseyinacacak-janea thanks for your work on helping to make the test more reliable.